src/main/java/fr/moribus/imageonmap/commands/maptool/UpdateCommand.java

```diff
@@ -83,50 +153,85 @@ protected void run() throws CommandException {
-                                        exception.getMessage());
-                            }
-                        });
-            } finally {
-                ActionBar.removeMessage(player);
+                url1 = new URL(url);
+                //TODO replace by a check of the load status.(if not loaded load the mapmanager)
+                MapManager.load(false);//we don't want to spam the console each time we reload the mapManager
+
+                Integer[] size = {1, 1};
+                if (map.getType() == ImageMap.Type.POSTER) {
+                    size = map.getSize(map.getUserUUID(), map.getId());
+                }
+                //assert size != null;
+                int width = size[0];
+                int height = size[1];
+                try {
+                    if (playerSender != null) {
+                        ActionBar.sendPermanentMessage(playerSender, ChatColor.DARK_GREEN + I.t("Updating..."));
+                    }
+                    ImageRendererExecutor
+                            .update(url1, scaling, uuid, map, width, height, new WorkerCallback<ImageMap>() {
+                                @Override
+                                public void finished(ImageMap result) {
+                                    if (playerSender != null) {
+                                        ActionBar.removeMessage(playerSender);
+                                        MessageSender.sendActionBarMessage(playerSender,
+                                                ChatColor.DARK_GREEN + I.t("The map was updated using the new image!"));
+                                    }
+                                }
+
+                                @Override
+                                public void errored(Throwable exception) {
+                                    if (playerSender != null) {
+                                        playerSender
+                                                .sendMessage(
+                                                        I.t("{ce}Map rendering failed: {0}", exception.getMessage()));
+                                    }
+                                    PluginLogger.warning("Rendering from {0} failed: {1}: {2}",
+                                            playerSender.getName(),
+                                            exception.getClass().getCanonicalName(),
+                                            exception.getMessage());
+                                }
+                            });
+                } finally {
+                    if (playerSender != null) {
+                        ActionBar.removeMessage(playerSender);
+                    }
+                }
+            } catch (MalformedURLException ex) {
+                if (playerSender != null) {
+                    warning(playerSender, I.t("Invalid URL."));
+                } else {
+                    PluginLogger.warning(I.t("Invalid URL."));
+                }
```

Comment:
Idem + `finalSender`

```suggestion
                warning(sender, I.t("Invalid URL."));
```